themes/default/templates/legacy.html

```diff
@@ -85,21 +98,55 @@ <h1>Legacy</h1>
       </form>
     </div>
 </div>
+
+<div class="row">
+  <div class="col-md-6">
+    <label>Filter from </label>
+    <input type="date" id="filter-date-start">
+    <label> to </label>
+    <input type="date" id="filter-date-end">
+    <button type="button" id="filter-date-reset" class="btn btn-default">Reset</button>
+  </div>
+  <div class="col-md-6">
+
+  </div>
+</div>
 {% endif %}
 
 {% endblock %}
 
 {% block javascript %}
   <script type="text/javascript">
+    // 1096614000 is the month Yelp was founded (Oct 2004). 3 extra zeros are added as milliseconds.
+    const defaultDateStart = new Date(1096614000000);
+
     const onViewLegacy = (e) => {
         e.preventDefault();
         const username = $('.leaderboard-filter :input').val();
-        if (username) window.location.href = '/legacy/' + username;
+        if ($('#filter-date-start').val() == null || $('#filter-date-start').val().length == 0 ||
+            $('#filter-date-end').val() == null || $('#filter-date-end').val().length == 0) {
```

Comment:
(nb) If `$('#filter-date-start').val()` is an empty string whenever it's not `null` (and thus the `.length` check), this line can be condensed to

```suggestion
        if (!$('#filter-date-start').val() || !$('#filter-date-end').val()) {
```

Reply:
Also can just consider calling `onDateFilterReset` whenever the legacy btn is clicked regardless of values.

Reply:
I used to let the filter dates reset when you click the legacy button, but I made it so that Jinja autofills the dates you filtered with back into the input fields. It's similar to how the username text box gets autofilled with whatever you searched for. Running `onDateFilterReset` would overwrite the Jinja date autofill, which is why I'm checking if the date inputs are empty (i.e., first navigating to the Legacy page) before calling `onDateFilterReset`.

Reply:
If you think we should get rid of the date autofill behavior, then let me know.